ytmusicapi/parsers/songs.py

```diff
@@ -21,7 +21,13 @@ def parse_song_artists_runs(runs: JsonList) -> JsonList:
     return artists
 
 
-def parse_song_runs(runs: JsonList) -> JsonDict:
+def parse_song_runs(runs: JsonList, api_result_types: list[str] = []) -> JsonDict:
```

Comment:
This is not so nice because we have to pass this information (api_result_types) through a method chain involving several functions. Can we do better or is there an alternative approach?